codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -44,55 +61,305 @@
     }
 
     @Override
-    protected void generateDocumentBodyShapeSerializers(GenerationContext context, Set<Shape> shapes) {
+    protected void generateOperationDocumentSerializer(
+            GenerationContext context, OperationShape operation
+    ) {
+        Model model = context.getModel();
+
```

Comment:
Nit: You can have a little less spacing between your Java code, the intent is just make sure your segments have logical groupings.